cmake/HPX_AddConfigTest.cmake

```diff
@@ -424,285 +242,6 @@ function(hpx_check_for_cxx11_std_atomic_128bit)
-  add_hpx_config_test(HPX_WITH_CXX11_OVERRIDE
-    SOURCE cmake/tests/cxx11_override.cpp
-    FILE ${ARGN}
-    CMAKECXXFEATURE cxx_override)
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx11_std_quick_exit)
-  add_hpx_config_test(HPX_WITH_CXX11_STD_QUICK_EXIT
-    SOURCE cmake/tests/cxx11_std_quick_exit.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_constexpr)
-  add_hpx_config_test(HPX_WITH_CXX14_CONSTEXPR
-    SOURCE cmake/tests/cxx14_constexpr.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_lambdas)
-  add_hpx_config_test(HPX_WITH_CXX14_LAMBDAS
-    SOURCE cmake/tests/cxx14_lambdas.cpp
-    FILE ${ARGN}
-    CMAKECXXFEATURE cxx_lambda_init_captures)
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_std_integer_sequence)
-  add_hpx_config_test(HPX_WITH_CXX14_INTEGER_SEQUENCE
-    SOURCE cmake/tests/cxx14_std_integer_sequence.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_std_is_final)
-  add_hpx_config_test(HPX_WITH_CXX14_IS_FINAL
-    SOURCE cmake/tests/cxx14_std_is_final.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_std_is_null_pointer)
-  add_hpx_config_test(HPX_WITH_CXX14_IS_NULL_POINTER
-    SOURCE cmake/tests/cxx14_std_is_null_pointer.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_std_result_of_sfinae)
-  add_hpx_config_test(HPX_WITH_CXX14_RESULT_OF_SFINAE
-    SOURCE cmake/tests/cxx14_std_result_of_sfinae.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_variable_templates)
-  add_hpx_config_test(HPX_WITH_CXX14_VARIABLE_TEMPLATES
-    SOURCE cmake/tests/cxx14_variable_templates.cpp
-    FILE ${ARGN}
-    CMAKECXXFEATURE cxx_variable_templates)
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_deprecated_attribute)
-  add_hpx_config_test(HPX_WITH_CXX14_DEPRECATED_ATTRIBUTE
-    SOURCE cmake/tests/cxx14_deprecated_attribute.cpp
-    FILE ${ARGN})
-endfunction()
-
-###############################################################################
-function(hpx_check_for_cxx14_return_type_deduction)
-  add_hpx_config_test(HPX_WITH_CXX14_RETURN_TYPE_DEDUCTION
-    SOURCE cmake/tests/cxx14_return_type_deduction.cpp
-    FILE ${ARGN}
-    CMAKECXXFEATURE cxx_return_type_deduction)
-endfunction()
```

Comment:
I'm not convinced we should remove the C++14 checks at this point. While we required C++11 for a while, we still had all the checks enabled to allow for the compilers to catch up. But please convince me!

Reply:
Fair enough, but it comes down to what compilers we're planning on supporting. The pycicle builds with the oldest gcc and clang, and nvcc/clang-cuda are happy (they are forced to C++14). Do you know if MSVC has problems with any of these? Intel should be fine with anything that's in the standard library (rather than compiler) since it uses libstc++. Some of the checks are completely unused. I promise to add back whichever checks are necessary if it turns out a compiler that we care about doesn't support them. I won't fight back if you'd rather have them around (it does give a bit of peace of mind...).

Lastly (I know this isn't a very strong argument) if a compiler by now does not support C++14 fully it's probably not worth the effort to support it (I realize that in this case we already had the effort to support such cases, and I'm removing it).